### PR TITLE
LUCENE-9444: Improve test coverage for TaxonomyFacetLabels

### PR DESCRIPTION
**Pull request by goankur**
# Description

This PR improves test coverage for `TaxonomyFacetLabels` ( added in [PR 1893](https://github.com/apache/lucene-solr/pull/1893/files) ) by exercising the API - `TaxonomyFacetLabels.nextFacetLabel(docId, facetDimension)` to fetch facet labels for a specific dimension.

# Solution

The solution enhances the test method - `TestTaxonomyFacetCounts.testRandom()` and does the following
  - Pick a dimension at random
  - Filter expected facet labels to retain entries for chosen dimension only.
  - Invoke the API - `TaxonomyFacetLabels.nextFacetLabel(docId, facetDimension)` to get facet labels for specific dimension.
  - Ensures that `expected` and `actual` facet labels match.

# Tests

  - `TestTaxonomyFacetCounts.testRandom()` is enhanced to improve test coverage for `TaxonomyFacetLabels`. 
  - `FacetTestCase` is refactored to add a `dimension` parameter to utility method `getAllTaxonomyFacetLabels(...)`

# Checklist

Please review the following and check all that apply:

- [x ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ x] I have created a Jira issue and added the issue ID to my pull request title.
- [ x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ x] I have developed this patch against the `master` branch.
- [x ] I have run `./gradlew check`.
- [x ] I have added tests for my changes.
- [x ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/facet/src/test/org/apache/lucene/facet/FacetTestCase.java [hunk 1]
```diff
@@ -70,32 +70,51 @@ public Facets getTaxonomyFacetCounts(TaxonomyReader taxoReader, FacetsConfig con
...
+   * @param dimension Retain facet labels for supplied dimension only. A null value fetches all facet labels.
+   * @param facetLabelReader {@FacetLabelReader} instance use to get facet labels for input docId.
+   * @return {@code List<FacetLabel>} containing matching facet labels.
+   * @throws IOException when a low-level IO issue occurs while reading facet labels.
+   */
+  List<FacetLabel> allFacetLabels(int docId, String dimension, FacetLabelReader facetLabelReader) throws IOException {
+    List<FacetLabel> facetLabels = new ArrayList<>();
+    FacetLabel facetLabel;
+    if (dimension != null) {
+      for (facetLabel = facetLabelReader.nextFacetLabel(docId, dimension); facetLabel != null; ){
```
Comment (mikemccand) [inline lucene/facet/src/test/org/apache/lucene/facet/FacetTestCase.java:105]: Missing space before `{`?

File: lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetCounts.java [hunk 2]
```diff
@@ -696,10 +696,9 @@ public void testRandom() throws Exception {
               } else {
                 expectedCounts[j].put(doc.dims[j], v.intValue() + 1);
               }
-
               // Add document facet labels
               facetLabels.add(new FacetLabel("dim" + j, doc.dims[j]));
-            }
+             }
```
Comment (mikemccand) [inline lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetCounts.java:701]: Hmm remove this added space?

**Review by mikemccand — APPROVED** [review]
Thanks for improving the test @goankur!

I left a couple small whitespace comments.

File: lucene/facet/src/test/org/apache/lucene/facet/FacetTestCase.java (hunk 1 above)
Comment (goankur) [reply lucene/facet/src/test/org/apache/lucene/facet/FacetTestCase.java]: Thanks @mikemccand.  This is fixed.

File: lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetCounts.java (hunk 2 above)
Comment (goankur) [reply lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetCounts.java]: Fixed.

**Review by mikemccand — APPROVED** [review]
Thanks @goankur

**Pull request merged**